### PR TITLE
chore: migrate license check to dependency review action

### PR DESCRIPTION
**Pull request by zchenwei**
#### Description of changes

The [dependency review](https://github.com/actions/dependency-review-action) action that we use to check vulnerabilities in the dependency graph now supports license checking in itself. We can migrate the existing license checking action over to it and remove the script maintained by us.

#### Description of how you validated changes

I sanity tested with the lovely `is-odd` package licensed under MIT.

- Action failed when MIT license was missing from allowlist - https://github.com/aws-amplify/amplify-ui/actions/runs/5360762238/jobs/9726031339
- Action succeeded when MIT license was on allowlist  - https://github.com/aws-amplify/amplify-ui/actions/runs/5360771187/jobs/9726053687

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/test-fork-prs.yml [hunk 1]
```diff
@@ -193,6 +193,50 @@ jobs:
         with:
           base-ref: ${{ needs.setup.outputs.base_sha }}
           head-ref: ${{ needs.setup.outputs.commit_id }}
+          allow-licenses:
```
Comment (ioanabrooks) [inline .github/workflows/test-fork-prs.yml:196]: Can we have a reusable action for dependency review, and list all these licenses in one place?

Comment (zchenwei) [reply .github/workflows/test-fork-prs.yml:196]: Talked with @wlee221 long time ago and we create a ticket to refactor this part into a reusable miscellaneous workflow.

Comment (zchenwei) [reply .github/workflows/test-fork-prs.yml:196]: Well, actually I found this could be placed into a config. Let me try when I have some time 😄

**Review by wlee221 — APPROVED** [review]

**Review by ioanabrooks — APPROVED** [review]
Nice!!

**Pull request merged**